scanpy/get/get.py

```diff
@@ -437,3 +437,141 @@ def _set_obs_rep(adata, val, *, use_raw=False, layer=None, obsm=None, obsp=None)
             "That was unexpected. Please report this bug at:\n\n\t"
             " https://github.com/theislab/scanpy/issues"
         )
+
+
+GroupsList = List[str]
+GroupsLists = List[GroupsList]
+GroupsDict = Dict[str, GroupsList]
+
+
+def split_by(
+    adata: AnnData,
+    key: str,
+    groups: Optional[Union[GroupsList, GroupsLists, GroupsDict]] = None,
```

Comment:
```suggestion
    key: str,
    *.
    groups: Optional[Union[GroupsList, GroupsLists, GroupsDict]] = None,
```

Could we everything past `key` keyword only? This way we don't have to worry about changing orders.